js/start.js

```diff
@@ -634,6 +635,20 @@ function start() {
               }
             });
           }
+
+          // Make sure the client is running on a compatible version of the server.
+          if (app.settings.prettyServerVer !== serverVersionRequired) {
+            const cLink = `<a href="https://github.com/OpenBazaar/openbazaar-desktop/releases">${app.polyglot.t('serverVersionWarning.clientLink')}</a>`;
+            const sLink = `<a href="https://github.com/OpenBazaar/openbazaar-go/releases">${app.polyglot.t('serverVersionWarning.serverLink')}</a>`;
+            const message = app.polyglot.t('serverVersionWarning.message', {
+              serverVersion: app.settings.prettyServerVer,
+              expectedVersion: serverVersionRequired,
+            });
+            const body = `<p>${message}</p><p>${cLink}</p><p>${sLink}</p>`;
+            openSimpleMessage(
+              app.polyglot.t('serverVersionWarning.title'), body
+            ).$el.css('z-index', '9999999');
```

Comment:
I guess this incredibly ugly z-index approach is ok if used extremely judiciously. Normally, you'd just want to open the modal in the right sequence for it to be on top. But... I understand some pages end up asynchronously loading a modal, so that could be a can of worms. The alternative would be to delay the router until they close this dialog, but I suppose it's not super critical here.